app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -70,70 +74,84 @@ class GroupConversationDetailsViewModel @Inject constructor(
         savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)!!
     )
 
-    var groupOptionsState: GroupConversationOptionsState by mutableStateOf(GroupConversationOptionsState(conversationId))
+    private val _groupOptionsState = MutableStateFlow(GroupConversationOptionsState(conversationId))
+    val groupOptionsState: StateFlow<GroupConversationOptionsState> = _groupOptionsState
+
     var requestInProgress: Boolean by mutableStateOf(false)
+        private set
 
     init {
         observeConversationDetails()
         checkIsSelfUserMember()
     }
 
     private fun observeConversationDetails() {
-        viewModelScope.launch {
-            observerSelfUser().first()
-                .also { selfUser ->
-                    // TODO(QOL): refactor to one usecase that return group info and members
-                    observeConversationMembers(conversationId)
-                        .map { it.isSelfAnAdmin }
-                        .distinctUntilChanged()
-                        .also { isSelfAdminFlow ->
-                            val conversationDetailsFlow = observeConversationDetails(conversationId)
-                                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>() // TODO handle StorageFailure
-                                .map { it.conversationDetails }
-                            combine(
-                                conversationDetailsFlow,
-                                isSelfAdminFlow,
-                                getSelfTeam()
-                            ) { conversationDetails, isSelfAnAdmin, selfTeam ->
-                                Triple(
-                                    conversationDetails,
-                                    isSelfAnAdmin,
-                                    selfTeam
+        viewModelScope.launch(dispatcher.io()) {
```

Comment:
Keep in mind that this can lead to random crashes as we update the composable states outside the Main thread.

See this, for example: https://github.com/wireapp/wire-android-reloaded/pull/959

We should:
- Use `flowOn(dispatcher)`
- _Prefereably_ inside the UseCase isntead of doing this on the ViewModels, so we can do in a single place instead of everywhere. Sometimes we will need to do this in the ViewModel to extract maximum performance.

Reply:
actually we are not updating composable states, but emit into StateFlow outside the Main thread, which is much better I think and prevent us from crash :)